src/controller/UserUpgradeController.php

```diff
@@ -126,14 +112,15 @@ public function displayUserUpgradePage(): void
      * 
      * @return void
      */
-    public function manageUserUpgrade(int $userId, string $role, string $isAllowed): void
+    public function manageUserUpgrade(int $userId, string $role, int $isAllowed): void
     {
         $template = RouteMapper::UserUpgradeView->getTemplate();
-        $this->_userService->modifyRole($userId, $role, intval($isAllowed));
+        $data = [];
+        $this->_userService->modifyRole($userId, $role, (int) $isAllowed);
```

Comment:
pas besoin de caster en int ici puisque tu le mets déjà dans ton paramètre de fonction : 
```suggestion
        // public function manageUserUpgrade(int $userId, string $role, int $isAllowed): void
        $this->_userService->modifyRole($userId, $role, $isAllowed);
```

Reply:
ok